tests/source/brsHamcrest/Test_brsHamcrest_Helpers.brs

```diff
@@ -190,3 +190,152 @@ sub test_IsEnumerable_false (t as Object)
 
     teardown_brsHamcrest_Helpers()
 end sub
+
+
+sub test_coreDoMatch_withIdenticalDataTypes (t as Object)
+    test = setup_brsHamcrest_Helpers
+
+    'GIVEN'
+    getFooObj = function () as Object
+        return {
+            knownString: "knownStringParam"
+            knownInteger: 1
+            knownBoolean: true
+            knownFloat: 1/3
+            knownDouble: 2.3#
+            knownLongInteger: 987654321&
+        }
+    end function
+    'WHEN'
+    result = coreDoMatch(getFooObj(), getFooObj())
+    'THEN'
+    t.assertTrue(result)
+
+    teardown_brsHamcrest_Helpers()
+end sub
+
+
+sub test_coreDoMatch_withDifferentDataTypes (t as Object)
+    test = setup_brsHamcrest_Helpers
+
+    'GIVEN'
+    getFooObj = function () as Object
+        return {
+            knownString: "knownString"
+            knownInteger: 1
+            knownBoolean: true
+        }
+    end function
+
+    getDifferentFooObj = function () as Object
+        return {
+            knownString: "knownString"
+            differentString: "differentString"
+            knownInteger: 1
+            knownBoolean: true
+        }
+    end function
+    'WHEN'
+    result = coreDoMatch(getFooObj(), getDifferentFooObj())
+    'THEN'
+    t.assertFalse(result)
+
+    teardown_brsHamcrest_Helpers()
+end sub
```

Comment:
Can we have a test that compares when two objects have the same keys but different values? Eg:
```
objectA = { value1: "a", value2: "b", value3: "c" }
objectB = { value1: "x", value2: "y", value3: "z" }
```